pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -554,7 +564,12 @@ private void buildDedupMeta(ImmutableSegmentImpl immutableSegment) {
     PartitionDedupMetadataManager partitionDedupMetadataManager =
         _tableDedupMetadataManager.getOrCreatePartitionManager(partitionGroupId);
     immutableSegment.enableDedup(partitionDedupMetadataManager);
-    partitionDedupMetadataManager.addSegment(immutableSegment);
+    SegmentDataManager oldSegmentManager = _segmentDataManagerMap.get(segmentName);
+    if (oldSegmentManager != null) {
+      partitionDedupMetadataManager.replaceSegment(oldSegmentManager.getSegment(), immutableSegment);
+    } else {
+      partitionDedupMetadataManager.addSegment(immutableSegment);
```

Comment:
not sure if you have logs inside, we can log msg here as in handleUpsert() when adding or replacing segments.

Reply:
yes, we have logs inside, but we can also add one more log here.